File: backend/portfolio_lek/models/Message.go
package models

import "gorm.io/gorm"

type Message struct {
	gorm.Model
	Name    string `gorm:"not null" json:"name"`
	Email   string `gorm:"unique;not null" json:"email"`
	Message string `json:"message"`
	// 	Role     string `gorm:"not null" json:"role"`
}

File: assets/js/message.js

import { API_URL } from './config.js';

window.sendMessage = function () {
    const name = $("#name").val();
    const email = $("#email").val();
    const message = $("#message").val();

    $.ajax({
        url: API_URL + "/Message", 
        type: "POST",
        contentType: "application/json",
        data: JSON.stringify({ name, email, message }),
        success: function (response) {
            alert("Message sent successfully!");
            console.log(response);
        },
        error: function (error) {
            alert("Failed to send message.");
            console.log(error);
        }
    });
};
